client/auth/auth-code-button.jsx

```diff
@@ -3,65 +3,114 @@
 /**
  * External dependencies
  */
-
 import React from 'react';
 import { localize } from 'i18n-calypso';
+import request from 'superagent';
+import { identity } from 'lodash';
 
 /**
  * Internal Dependencies
  */
-import { requestCode, resetCode } from 'lib/auth-code-request-store/actions';
-import { default as Store, requestState } from 'lib/auth-code-request-store';
 import Notice from 'components/notice';
 
-class AuthCodeButton extends React.Component {
-	state = Store.get();
+const initialState = { status: 'ready', errorLevel: false, errorMessage: false };
+
+export class AuthCodeButton extends React.Component {
+	static defaultProps = {
+		translate: identity,
+	};
+
+	state = initialState;
 
 	componentDidMount() {
-		Store.on( 'change', this.refreshData );
+		this.mounted = true;
 	}
 
 	componentWillUnmount() {
-		Store.off( 'change', this.refreshData );
+		this.mounted = false;
+		clearTimeout( this.timeout );
 	}
 
-	refreshData = () => {
-		this.setState( Store.get() );
+	// Reset the notice (which might be in 'error' or 'completed' state) back to 'ready' state.
+	// Happens either when error notice is dismissed, or when 30s timeout elapses.
+	resetSMSCode = () => {
+		clearTimeout( this.timeout );
+		this.timeout = null;
+		this.setState( initialState );
 	};
 
-	requestSMSCode = e => {
-		e.preventDefault();
-		requestCode( this.props.username, this.props.password );
+	requestSMSCode = () => {
+		this.setState( { status: 'requesting' } );
+
+		return request
```

Comment:
if we stash a ref to the request (the result of everything up to the `then`), we can call `request.abort()` to avoid handling the request. That would allow us to drop the `isMounted` tracking elsewhere in this component.

Reply:
Nice idea! I reimplemented the request as suggested: it aborts on unmount. Tracking the `mounted` flag is no longer needed.

I also fixed a little UI regression by putting back the `href="#"` attribute to the notice/button/link. It turns out it's necessary to trigger a `cursor:pointer` style on the element -- it's present only for elements that match `a:any-link`.

I added testing instructions to the PR description.